lib/pluginmanager/proxy_support.rb

```diff
@@ -54,27 +54,37 @@ def extract_proxy_values_from_uri(proxy_uri)
   }
 end
 
+def parse_proxy_string(proxy_string)
+  proxy_uri = URI.parse(proxy_string)
+  if !proxy_uri.kind_of?(URI::HTTP) and !proxy_uri.kind_of?(URI::HTTPS)
```

Comment:
nitpick: 

use `&&` instead of `and` in conditionals; this is a surprising gotcha in Ruby because they are _mostly_ interchangeable, but carry different order precedence that can be very surprising and hard to debug when we get to more complex expressions.

More reading: https://stackoverflow.com/questions/1426826/difference-between-and-and-in-ruby

Reply:
Good to know, thank you!